File: 0_3.leetcode/java/Longest_Common_Prefix.java
public class Longest_Common_Prefix {
    public String longestCommonPrefix(String[] strs) {
        return find(strs);
    }

    private String find(String[] strs) {
        StringBuilder sb = new StringBuilder();
        int len = strs[0].length();
        for (int i=0;i<len;i++) {
            char prefix = strs[0].charAt(i);
            for (int j=1;j<strs.length;j++) {
                if (strs[j].length() == i || strs[j].charAt(i) != prefix) return sb.toString();
            }

            sb.append(prefix);
        }

        return sb.toString();
    }
}

File: 0_3.leetcode/kotlin/src/main/kotlin/Longest_Common_Prefix.kt
fun main() {
    val result =  longestCommonPrefix(arrayOf("flower", "float", "fly"))
    println("result = $result")
}

fun longestCommonPrefix(strs: Array<String?>?): String? {
    val first = strs!![0]!!
    val sb = StringBuilder()
    first.forEachIndexed { idx, prefix ->
        for (i in 1 until  strs.size) {
            if (strs[i]!!.length == idx || strs[i]!![idx] != prefix) {
                return sb.toString()
            }
        }
        sb.append(prefix)
    }

    return sb.toString()
}